types/src/util.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use rlp::{Decodable, DecoderError, Encodable, RlpStream, UntrustedRlp};
+
+use unexpected::Mismatch;
+
+pub fn encode_mismatch<'a, T>(mismatch: &Mismatch<T>, s: &'a mut RlpStream) -> &'a mut RlpStream
```

Comment:
Move `Mismatch` to `types` package and implement `Encodable` and `Decodable` traits there.